crates/matrix-sdk-ui/src/timeline/inner.rs

```diff
@@ -144,6 +168,119 @@ impl<P: RoomDataProvider> TimelineInner<P> {
         state.items.subscribe_filter_map(f)
     }
 
+    pub(super) async fn toggle_reaction_local(
+        &self,
+        annotation: &Annotation,
+    ) -> Result<ReactionAction, super::Error> {
+        let mut state = self.state.lock().await;
+
+        let txn_id = TransactionId::new();
```

Comment:
nit: can you inline it where it's used? It's a bit further down, and there's at least one flow in which it's unused.